### PR TITLE
[chore] Change processor config tests to unmarshal config only for that component. (part 1)

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Change processor config tests to unmarshal config for that component.

- [x] attributesprocessor
- [x] cumulativetodeltaprocessor
- [x] deltatorateprocessor
- [x] filterprocessor
- [x] groupbyattrsprocessor
- [x] groupbytraceprocessor

**Link to tracking Issue:** <Issue number if applicable>

Updates https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

### CODE REVIEW

**Comment by kovrus** [comment]
Some checks fail coz of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13325

File: processor/cumulativetodeltaprocessor/factory_test.go [hunk 1]
```diff
@@ -44,49 +46,42 @@ func TestCreateDefaultConfig(t *testing.T) {
...
-		factory := NewFactory()
-		factories.Processors[typeStr] = factory
-		config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-		assert.NoError(t, err)
+			var id string
+			parts := strings.Split(k, "/")
+			if len(parts) > 1 {
+				id = parts[1]
+			}
+			sub, err := cm.Sub(config.NewComponentIDWithName(typeStr, id).String())
```
Comment (bogdandrutu) [inline processor/cumulativetodeltaprocessor/factory_test.go:65]: Use ComponentID.UnmarshalText()

File: processor/cumulativetodeltaprocessor/config_test.go [hunk 2]
```diff
@@ -71,59 +63,62 @@ func TestLoadingFullConfig(t *testing.T) {
...
-			config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-			if test.succeed {
-				assert.NotNil(t, config)
-				assert.NoError(t, err)
+			if tt.expected != nil {
+				assert.NoError(t, cfg.Validate())
+				assert.Equal(t, tt.expected, cfg)
 			} else {
-				assert.EqualError(t, err, fmt.Sprintf("processor %q has invalid configuration: %s", typeStr, test.errorMessage))
+				assert.EqualError(t, cfg.Validate(), tt.errorMessage)
```
Comment (bogdandrutu) [inline processor/cumulativetodeltaprocessor/config_test.go:121]: I like this way more (that you also use in a different test):
```golang
if tt.expected == nil {
	assert.EqualError(t, cfg.Validate(), tt.errorMessage)
	return
}
assert.NoError(t, cfg.Validate())
assert.Equal(t, tt.expected, cfg)
```

File: processor/deltatorateprocessor/config_test.go [hunk 3]
```diff
@@ -15,83 +15,58 @@
...
-func TestLoadingFullConfig(t *testing.T) {
+func TestLoadConfig(t *testing.T) {
+	t.Parallel()
+
 	tests := []struct {
-		configFile string
-		expCfg     *Config
+		id           config.ComponentID
+		expected     config.Processor
+		errorMessage string
```
Comment (bogdandrutu) [inline processor/deltatorateprocessor/config_test.go:34]: `errorMessage` not configured, remove and code associated with it.

File: processor/deltatorateprocessor/factory_test.go [hunk 4]
```diff
@@ -44,49 +46,42 @@ func TestCreateDefaultConfig(t *testing.T) {
...
-		factory := NewFactory()
-		factories.Processors[typeStr] = factory
-		config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-		assert.NoError(t, err)
+			var id string
+			parts := strings.Split(k, "/")
+			if len(parts) > 1 {
+				id = parts[1]
+			}
+			sub, err := cm.Sub(config.NewComponentIDWithName(typeStr, id).String())
```
Comment (bogdandrutu) [inline processor/deltatorateprocessor/factory_test.go:65]: Same use `ComponentID.UnmarshalText`. But not sure you need to do that, just simply `sub, err := cm.Sub(k)`

File: processor/filterprocessor/factory_test.go [hunk 5]
```diff
@@ -80,24 +83,39 @@ func TestCreateProcessors(t *testing.T) {
...
-				tp, tErr := factory.CreateTracesProcessor(context.Background(), componenttest.NewNopProcessorCreateSettings(), cfg, consumertest.NewNop())
-				mp, mErr := factory.CreateMetricsProcessor(context.Background(), componenttest.NewNopProcessorCreateSettings(), cfg, consumertest.NewNop())
-				if strings.Contains(test.configName, "traces") {
-					assert.Equal(t, test.succeed, tp != nil)
-					assert.Equal(t, test.succeed, tErr == nil)
+				var id string
+				parts := strings.Split(k, "/")
+				if len(parts) > 1 {
+					id = parts[1]
+				}
```
Comment (bogdandrutu) [inline processor/filterprocessor/factory_test.go:100]: Same comment about the split.

File: processor/deltatorateprocessor/config_test.go (hunk 3 above)
Comment (kovrus) [reply processor/deltatorateprocessor/config_test.go:34]: It is provided in one of the test cases https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13298/files#diff-8ecc2408749d30e707b9620e602b6cfceb1cfc30867e9f7eaaaf1f14f46f1d64R48

File: processor/deltatorateprocessor/factory_test.go (hunk 4 above)
Comment (kovrus) [reply processor/deltatorateprocessor/factory_test.go]: `sub, err := cm.Sub(k)` that is enough

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**